tests/fast-integration/eventing-test/eventing-test.js

```diff
@@ -223,52 +224,78 @@ describe('Eventing tests', function() {
-  // It also compares the consumers creation timestamp before and after upgrade
-  // and if the timestamp is the same, we conclude that the consumer is not re-created.
-  function jsStreamAndConsumersNotReCreatedTestSuite() {
-    context('stream and consumer not re-created check with NATS backend', function() {
-      let preUpgradeStreamData = null;
-      let preUpgradeConsumersData = null;
-
-      before('fetch eventing-js-test data configMap', async function() {
-        debug(`fetch configMap: ${jetStreamTestConfigMapName}...`);
-        const cm = await getConfigMapWithRetries(jetStreamTestConfigMapName, testNamespace);
-        if (!cm || !isUpgradeJob) {
-          debug(`Skipping stream and consumers not re-created check`);
-          this.skip();
-          return;
-        }
+      context('Pre-upgrade tasks to save stream and consumer data', function() {
+        before('Check if this is pre-upgrade stage', async function() {
+          if (upgradeStage !== 'pre') {
+            debug(`Skipping pre-upgrade tasks...`);
+            this.skip();
+          }
+          debugBanner('Pre-upgrade tasks to save stream and consumer data');
+        });
 
-        // if configMap is found, then check for re-creation
-        expect(cm.data).to.have.nested.property('stream');
-        expect(cm.data).to.have.nested.property('consumers');
-        preUpgradeStreamData = JSON.parse(cm.data.stream);
-        preUpgradeConsumersData = JSON.parse(cm.data.consumers);
+        it('saving JetStream data to test stream and consumers will not be re-created ' +
+            'by kyma upgrade', async function() {
+          debug(`Using NATS host: ${natsApiRuleVSHost}`);
+          await saveJetStreamDataForRecreateTest(natsApiRuleVSHost, jsRecreatedTestConfigMapName);
+        });
       });
 
-      it('upgrade should not have re-created stream', async function() {
-        debug('Verifying that the stream was not re-created by the kyma upgrade...');
-        await checkStreamNotReCreated(natsApiRuleVSHost, preUpgradeStreamData);
-      });
+      context('Post-upgrade tasks to verify that stream and consumer not re-created ' +
+          'check with NATS backend', function() {
+        let preUpgradeStreamData = null;
+        let preUpgradeConsumersData = null;
+
+        before('Check if this is post-upgrade stage', async function() {
+          if (upgradeStage !== 'post') {
+            debug(`Skipping post-upgrade tasks...`);
+            this.skip();
+          }
+          debugBanner('Post-upgrade tasks to verify that stream and consumer not re-created check with NATS backend');
+        });
 
-      it('upgrade should not have re-created consumers', async function() {
-        debug('Verifying that the consumers were not re-created by the kyma upgrade...');
-        await checkConsumerNotReCreated(natsApiRuleVSHost, preUpgradeConsumersData);
+        before('fetch data from configMap', async function() {
```

Comment:
nitpick:  it is better to use `fetch data stream and consumer data from configMap`

Reply:
updated